server.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include <string>
+#include <boost/bind.hpp>
+#include <boost/shared_ptr.hpp>
+#include <boost/enable_shared_from_this.hpp>
+#include <boost/asio.hpp>
+
+using boost::asio::ip::tcp;
+
+namespace {
+  const int PORT = 8118;
+}
+
+// shared_ptr and enable_shared_from_this keeps the connection alive
+// as long as there's an operation referring to it.
+class tcp_connection
+    :  public boost::enable_shared_from_this<tcp_connection> {
+  public:
+  	typedef boost::shared_ptr<tcp_connection> pointer;
+
+  	static pointer create(boost::asio::io_service& io_service) {
+  	  return pointer(new tcp_connection(io_service));
+  	}
+
+  	tcp::socket& socket() {
+  	  return socket_;
+  	}
+
+    // async_write() serves data to the client
+  	void start() {
+  	  // data meant to be sent is stored in message_
+   	  message_ = "HTTP-Version: HTTP/1.0 200 OK\nContent-Type: text/plain";
+   	             // Later we will need to add the HTTP request
+
+
+      boost::asio::async_write(socket_, boost::asio::buffer(message_),
+      	  boost::bind(&tcp_connection::handle_write, shared_from_this(),
+      	  	  boost::asio::placeholders::error,
+      	  	  boost::asio::placeholders::bytes_transferred));
+  	}
+
+  private:
+  	tcp_connection(boost::asio::io_service& io_service) : socket_(io_service) {}
+
+  	void handle_write(const boost::system::error_code& error,
+  	    size_t bytes_transferred) {}
+
+    tcp::socket socket_;
+    std::string message_;
+};
+
+class tcp_server {
+  public:
+  	tcp_server (boost::asio::io_service& io_service)
+  	    : acceptor_(io_service, tcp::endpoint(tcp::v4(), 8118)) {
```

Comment:
you're right - I was playing with it during testing and never changed it.. so I've changed it to PORT